### ISSUE TITLE
VerthashSP miner issues

### ISSUE THREAD

**Issue by papadrams** [RainbowMiner/RainbowMiner#2174]
Good Morning,

VerthashSP was always failing in my machines and i didnt give it any attention. I tried to see why today.
VerthashSP.ps1 is a mess!! Instead of verthashsp it has links to teamblack miner, while the commands are for verthashsp!!

**Comment by papadrams** [comment]
Also both verthash miner and verthashsp miner are running in the background. No windows to be seen!

**Comment by RainbowMiner** [comment]
Both are ok - the verhashSP comes with a Teamblack distribution. And since the miner doesn't have an API for us to connect, it needs to be run in background to catch and parse the screen output.

**Comment by papadrams** [comment]
Ok, then i need to search further. I can run it both on nvidia and amd outside of rbm, but it fails in rbm

**Comment by RainbowMiner** [comment]
You can send me a current debug file, maybe I can see something in the logs.

**Comment by papadrams** [comment]
I did some testing with no result so far. I have put back the original .ps1 and let it faile the benchmarks to get a debug file.
You will see that for the amd card it uses --cl-devices 0   . In theory it should be using --cl-devices 1
0 is the intel apu.
The file is in your mail.

**Comment by RainbowMiner** [comment]
> You will see that for the amd card it uses --cl-devices 0 . In theory it should be using --cl-devices 1
> 0 is the intel apu.

That for sure is the explanation. The module needs another GPU count for OpenCL. I'll change this.

**Comment by papadrams** [comment]
that doesnt explain nvidia though

**Comment by RainbowMiner** [comment]
It is either, that the miner doesn't run stable on the 1060 (in fact, it starts to mine, but then crashes) or that it has bad file access handling. Two instances, that try to read the one verthash.dat file may be too much. Or two instances of the miner in parallel, even if the one crashes almost at once may be too much. Maybe it's just best to bury the old thing.

I am also not sure, how that miner counts the opencl devices. All other miners say: 0=AMD, 1=INTEL, but this VerthashSP tells another story. I bet it counts, sorted by PCIe bus id.

**Comment by papadrams** [comment]
Yeah, you are propably right. Bury it
If i wanna make a recommendation, should i create an issue or a discussion?

**Comment by RainbowMiner** [comment]
Better open an issue.

**Comment by papadrams** [comment]
Thanks!
